src/main/java/com/epam/esm/util/validation/dao/TagValidationDao.java

```diff
@@ -0,0 +1,41 @@
+package com.epam.esm.util.validation.dao;
+
+import com.epam.esm.persistence.dao.TagDao;
+import com.epam.esm.util.validation.BeforeTagValidation;
+import com.epam.esm.web.exception.EntityBadInputException;
+import com.epam.esm.web.exception.EntityNotFoundException;
+import org.springframework.stereotype.Component;
+
+import java.util.Optional;
+
+@Component
+public class TagValidationDao implements BeforeTagValidation<TagDao, Long> {
+
+    @Override
+    public void onBeforeDelete(Boolean isPresent, Long id) throws EntityNotFoundException {
```

Comment:
This piece of logic is too small to have it in the separate method. Do this check in the "delete" method itself.